x-pack/docs/en/security/authentication/jwt-realm.asciidoc

```diff
@@ -344,11 +346,100 @@ If realm `jwt2` successfully authenticates a client with a JWT for principal
 this defined role mapping, the realm can also look up this role mapping rule
 linked to realm `native1`.
 
+[[jwt-realm-runas]]
+===== Applying the `run_as` privilege to JWT realm users
+
+You can also apply the <<run-as-privilege-apply,`run_as` privilege>> to JWT
+realm users. If a user's roles in the JWT realm mapping contains the `run_as`
```

Comment:
I am not sure if this concept is clear in the text.

A JWT user can get roles via role mapping or delegated authorization. Both methods are valid for getting JWT user roles. Only one or the other can be configured in a realm.

No matter which option is used, if the JWT user's roles contain the `run_as` privilege, then authenticated requests can optionally switch context to `run_as` a different user. The text implies this is only possible for role mapping, but it can actually work with roles found via delegated authorization too.

As an example, a single request can change context twice.
1. JWT realm parses the username.
2. Delegated authorization takes the username, and does a different lookup than JWT realm.
3. Run-as takes the username from the request, and replaces both the username and roles.

Reply:
I'll modify this introductory paragraph to read as:

> {es} can retrieve roles for a JWT user through either role mapping or 
> delegated authorization. Regardless of which option you choose, you can apply the
> <<run-as-privilege-apply,`run_as` privilege>> to a role so that a user can
> submit authenticated requests to "run as" a different user. To submit requests as
> another user, include the `es-security-runas-user` header in your requests.
> Requests run as if they were issued from that user and {es} uses their roles.